src/webextension/background/message-ports.js

```diff
@@ -27,46 +27,85 @@ export default function initializeMessagePorts() {
 
   browser.runtime.onMessage.addListener(async (message, sender) => {
     switch (message.type) {
+    case "account_status":
+      return {account: getAccount().status()};
     case "open_view":
       return openView(message.name).then(() => ({}));
     case "close_view":
       return closeView(message.name).then(() => ({}));
 
-    case "signin":
-      return getAuthorization().signIn(message.interactive);
     case "initialize":
-      return openDataStore().then(async (ds) => {
-        await ds.initialize({
-          password: message.password,
-        });
-        await saveAuthorization(browser.storage.local);
-        await updateBrowserAction(ds);
+      return openDataStore().then(async (datastore) => {
+        await datastore.initialize();
+        // FIXME: be more implicit on saving account info
+        await accounts.saveAccount(browser.storage.local);
+        await updateBrowserAction({datastore});
+        if (!message.silent) {
+          await openView("manage");
+        }
+
+        return {};
+      });
+    case "upgrade":
```

Comment:
`"upgrade"` doesn't seem like the right name for this, but I'm not sure what the best name would be. Maybe something like `"link_fxa"`?

Reply:
I went back and forth and ended up with "upgrade" due to my reading of the latest copy.  How about "bind_account"?